src/main.cpp

```diff
@@ -35,136 +35,118 @@
 #include "Snapshotter.hpp"
 #include "TopicFilter.hpp"
 
-#include <ros/ros.h>
-#include <snapshotter/TakeSnapshot.h>
+#include <rclcpp/executors.hpp>
+#include <rclcpp/node.hpp>
+#include <snapshotter/srv/take_snapshot.hpp>
 
 #include <mutex>
 #include <string>
 #include <vector>
 
 using namespace snapshotter;
 
-BagCompression getCompression(ros::NodeHandle& nh)
+BagCompression getCompression(rclcpp::Node& nh)
 {
-    std::string compression;
-    if (!nh.getParam("bag_compression", compression))
-    {
-        throw std::runtime_error("Parameter 'bag_compression' missing");
-    }
+    std::string compression = nh.get_parameter("bag_compression").as_string();
 
     if (compression == "none")
+    {
         return BagCompression::NONE;
+    }
     if (compression == "slow")
+    {
         return BagCompression::SLOW;
+    }
     if (compression == "fast")
+    {
         return BagCompression::FAST;
+    }
     throw std::runtime_error("Parameter bag_compression has illegal value: '" + compression + "'");
 }
 
 int main(int argc, char** argv)
 {
-    ros::init(argc, argv, "snapshotter_test");
-    ros::NodeHandle nh("~");
+    rclcpp::init(argc, argv);
 
-    std::vector<std::string> excludeRegexes;
-    if (!nh.getParam("exclude_topics", excludeRegexes))
-    {
-        throw std::runtime_error("Parameter 'exclude_topics' missing");
-    }
-    std::vector<std::string> includeRegexes;
-    if (!nh.getParam("include_topics", includeRegexes))
-    {
-        throw std::runtime_error("Parameter 'include_topics' missing");
-    }
+    rclcpp::NodeOptions ops;
+    ops.automatically_declare_parameters_from_overrides(true);
+
+    rclcpp::Node nh("snapshotter", ops);
+
+    std::vector<std::string> excludeRegexes = nh.get_parameter("exclude_topics").as_string_array();
+    std::vector<std::string> includeRegexes; // = nh.get_parameter("include_topics").as_string_array();
     TopicFilter topicFilter(excludeRegexes, includeRegexes);
 
     Snapshotter::Config cfg;
 
-    int maxMemoryMb = 0;
-    if (!nh.getParam("max_memory_mb", maxMemoryMb))
-    {
-        throw std::runtime_error("Parameter 'max_memory_mb' missing");
-    }
+    int maxMemoryMb = nh.get_parameter("max_memory_mb").as_int();
     cfg.maxMemoryBytes = size_t(maxMemoryMb) * size_t(1024 * 1024);
```

Comment:
```suggestion
    size_t maxMemoryMb = static_cast<size_t>(nh.get_parameter("max_memory_mb").as_int());
    cfg.maxMemoryBytes = maxMemoryMb * 1024 * 1024;
```

Reply:
needs to stay int, as I need to check for negativ input values